cmd/server/cadence/server_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// +build !race
+
+package cadence
+
+import (
+	"log"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/config"
+	_ "github.com/uber/cadence/common/persistence/nosql/nosqlplugin/cassandra"              // needed to load cassandra plugin
+	_ "github.com/uber/cadence/common/persistence/nosql/nosqlplugin/cassandra/gocql/public" // needed to load the default gocql client
+	"github.com/uber/cadence/common/service"
+	"github.com/uber/cadence/tools/cassandra"
+)
+
+type ServerSuite struct {
+	*require.Assertions
+	suite.Suite
+}
+
+func TestServerSuite(t *testing.T) {
+	suite.Run(t, new(ServerSuite))
+}
+
+func (s *ServerSuite) SetupTest() {
+	s.Assertions = require.New(s.T())
+}
+
+/*
+TestServerStartup tests the startup logic for the binary. When this fails, you should be able to reproduce by running "cadence-server start"
+*/
+func (s *ServerSuite) TestServerStartup() {
+	// If you want to test it locally, change it to false
+	runInBuildKite := true
+
+	env := "development"
+	zone := ""
+	rootDir := "../../../"
+	configDir := ConstructPathIfNeed(rootDir, "config")
+
+	log.Printf("Loading config; env=%v,zone=%v,configDir=%v\n", env, zone, configDir)
+
+	var cfg config.Config
+	err := config.Load(env, configDir, zone, &cfg)
+	if err != nil {
+		log.Fatal("Config file corrupted.", err)
+	}
+	// replace local host to docker network
+	if runInBuildKite {
+		ds := cfg.Persistence.DataStores[cfg.Persistence.DefaultStore]
+		ds.NoSQL.Hosts = "cassandra"
```

Comment:
I wonder if this should be taken from env variable? I see `CASSANDRA_HOST` being set buildkite pipeline.